lib/plug/builder.ex

```diff
@@ -241,4 +247,21 @@ defmodule Plug.Builder do
       end
     end
   end
+
+  defp log_halt(plug_type, plug, builder_opts) do
+    if level = builder_opts[:log_on_halt] do
+      quote do
+        plug = unquote(plug)
+        message = case unquote(plug_type) do
```

Comment:
This should be moved to outside of the quote. There is no need to execute this in the user code. :)